docs/Blog-Post.md

```diff
@@ -0,0 +1,276 @@
+[BioMedLM](https://crfm.stanford.edu/2022/12/15/biomedlm.html) last autumn.
+
+In the Jax community, there are a number of libraries popping up. Google has released [T5X](https://github.com/google-research/t5x)
+and [MaxText](https://github.com/google/maxtext). There are also a number of independent libraries, including [EasyLM](https://github.com/young-geng/EasyLM)
+and [JaxSeq](https://github.com/Sea-Snell/JAXSeq), both of which are based on [Flax](https://github.com/google/flax/)
+and modified libraries from [Hugging Face Transformers](https://github.com/huggingface/transformers/). Salesforce has
+released the Haiku-based [Jaxformer](https://github.com/salesforce/jaxformer).
+
+Despite the wide array of existing frameworks, when we started, we found that none of them fully addressed our needs.
+At CRFM, we wanted to achieve four goals:
+
+* **Legibility**: We wanted to be able to write code that was easy to read and understand, and that could be composed easily.
+* **Reproducibility**: We wanted to be able to reproduce our results exactly, even in the presence of preemption and
+  restarting from checkpoints.
+* **Scalability**: We wanted to be able to fully utilize the Cloud TPU resources we were given as part of Google's TPU Research Cloud program, as well as our own GPU resources.
+* **Efficiency**: We wanted to achieve the other three goals without sacrificing (much) efficiency.
+
+We chose [Jax](https://github.com/google/jax/) as our framework because it is a powerful, flexible, and performant,
+and offers strong reproducibility guarantees. Jax works well on TPUs, while we found that PyTorch support was still uneven.
+Jax is also a natural choice because it allows you to focus on the "what" of your code, and not on the "how:" details of
+partitioning and communication can be left to the XLA compiler. Finally, Jax makes reproducibility easy, since it uses
+bitwise deterministic PRNGs by default, with careful control over the PRNG state.
+
+However, Jax is a low-level framework, and we found that, by itself, it did not provide the legibility that we wanted.
+
+For us, legibility is a top concern. We therefore created two new libraries: **Haliax** and **Levanter**. Haliax is a
+named tensor library that focuses on improving the legibility and compositionality of deep learning code while still
+maintaining efficiency and scalability. Levanter is a library for training foundation models built on top of Haliax. In
+addition to the goals of legibility, efficiency, and scalability, Levanter further strives for bitwise reproducibility,
+meaning that the same code with the same data will produce the exact same result, even in the presence of preemption and
+restarts.
+
+## Haliax: Legibility via Named Tensors
+
+Haliax is a Jax library for building neural networks with named tensors, built on Jax and [Equinox](https://github.com/patrick-kidger/equinox),
```

Comment:
Seems like named tensors should be independent of neural networks?  Is there another layer of abstraction missing?

Reply:
I went back and forth on this. The things in Haliax are: named tensor, NN Modules that work with named tensors, and versions of jax primitives that transform nad distribute named tensors and module hierachies thereof. I think it's a pretty natural grouping. The NN part is pretty small.